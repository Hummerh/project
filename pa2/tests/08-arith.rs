fn main() {
      let a : i32 = 55;
      let b : i32 = -22;

      1 < 2 || 2 > -3 || a <= b || a >= (-b + 2) || (1 + 2) == (3 * 4);
      1 + 2 - (3 * 4) / 6 % 7;
      -1 + -2 - (-3 * -4) / -6 % -7;
      (-a + -b - -a) * -b / -a % -b;
      1 == 2 || a != b;
}
