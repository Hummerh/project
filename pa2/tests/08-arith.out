(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (a)))
                    (type-i32)
                    (lit-dec))
                (let
                    (pat-id
                        (id
                            (b)))
                    (type-i32)
                    (neg
                        (lit-dec)))
                (or
                    (or
                        (or
                            (or
                                (lt
                                    (lit-dec)
                                    (lit-dec))
                                (gt
                                    (lit-dec)
                                    (neg
                                        (lit-dec))))
                            (leq
                                (id
                                    (a))
                                (id
                                    (b))))
                        (geq
                            (id
                                (a))
                            (add
                                (neg
                                    (id
                                        (b)))
                                (lit-dec))))
                    (eq
                        (add
                            (lit-dec)
                            (lit-dec))
                        (mul
                            (lit-dec)
                            (lit-dec))))
                (sub
                    (add
                        (lit-dec)
                        (lit-dec))
                    (rem
                        (div
                            (mul
                                (lit-dec)
                                (lit-dec))
                            (lit-dec))
                        (lit-dec)))
                (sub
                    (add
                        (neg
                            (lit-dec))
                        (neg
                            (lit-dec)))
                    (rem
                        (div
                            (mul
                                (neg
                                    (lit-dec))
                                (neg
                                    (lit-dec)))
                            (neg
                                (lit-dec)))
                        (neg
                            (lit-dec))))
                (rem
                    (div
                        (mul
                            (sub
                                (add
                                    (neg
                                        (id
                                            (a)))
                                    (neg
                                        (id
                                            (b))))
                                (neg
                                    (id
                                        (a))))
                            (neg
                                (id
                                    (b))))
                        (neg
                            (id
                                (a))))
                    (neg
                        (id
                            (b))))
                (or
                    (eq
                        (lit-dec)
                        (lit-dec))
                    (neq
                        (id
                            (a))
                        (id
                            (b))))))))